### PR TITLE
clickhouse: set default logging level to error

### PR DESCRIPTION
**Pull request by sephii**
###### Description of changes

The current logging level of clickhouse is `trace`, which is *very* verbose (dozens of messages per second) and useless in most cases.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kirillrdy — APPROVED** [review]
works for me, not sure why trace is default, I had a look though clichouse git history and it looks like it was trace for a while

**Comment by cpcloud** [comment]
Lgtm

**Review by cpcloud — APPROVED** [review]

**Comment by Artturin** [comment]
Please create a issue upstream asking them to lower the default logging level or clarify why it's that by default

https://github.com/ClickHouse/ClickHouse/blob/14d6373bc75b548e13d8f48dead3c8048ed19776/programs/server/config.xml#L7-L24

File: pkgs/servers/clickhouse/default.nix [hunk 1]
```diff
@@ -56,6 +56,8 @@ stdenv.mkDerivation rec {
       $out/etc/clickhouse-server/config.xml
     substituteInPlace $out/etc/clickhouse-server/config.xml \
       --replace "<errorlog>/var/log/clickhouse-server/clickhouse-server.err.log</errorlog>" "<console>1</console>"
+    substituteInPlace $out/etc/clickhouse-server/config.xml \
+      --replace "<level>trace</level>" "<level>error</level>"
```
Comment (SuperSandro2000) [inline pkgs/servers/clickhouse/default.nix:60]: Why not info or warning?

**Comment by happysalada** [comment]
happy to help you merge this. Would you mind changing to a default level of warning ?

**Comment by sephii** [comment]
@happysalada I’ve changed to warning and [opened an issue upstream.](https://github.com/ClickHouse/ClickHouse/issues/49357)

**Comment by sephii** [comment]
Answer from upstream is they won’t change the default log level: https://github.com/ClickHouse/ClickHouse/pull/48452#issuecomment-1506986107

**Comment by den-crane** [comment]
Actually you are worrying about the wrong place. From my perspective the system database 10000% times nosier.

https://kb.altinity.com/altinity-kb-setup-and-maintenance/altinity-kb-system-tables-eat-my-disk/

**Pull request merged**